src/gromo/growing_module.py

```diff
@@ -636,8 +639,18 @@ def extended_forward(
             pre_activity -= linear_factor * self.optimal_delta_layer(x)
 
         if self.extended_input_layer:
-            assert x_ext is not None, "x_ext must be provided."
+            if x_ext is None:
+                raise ValueError(
+                    f"x_ext must be provided got None for {self.name}."
+                    f"As the input is extended, an extension is needed."
+                )
             pre_activity += sqrt_factor * self.extended_input_layer(x_ext)
+        else:
+            if x_ext is not None:
+                raise ValueError(
+                    f"x_ext must be None got {x_ext} for {self.name}."
+                    f"As the input is not extended, no extension is needed."
+                )
```

Comment:
It is preferable if this is a warning instead of an error. The user might not always know if there is an extended input and just pass the result of the previous module blindly. This would need extra checks that complicate the forward function

Reply:
I agree that this could be replaced by a warning. I thought that this behavior is very likely to be not desired and lead to a bug later so I directly raised an error. If you you convinced it should be a warning we can change it.

Reply:
I do not see how it can be harmful and the exception overcomplicates the forward function. I suggest a warning